pkg/controller/node/controller_utils.go

```diff
@@ -275,58 +267,51 @@ func recordNodeStatusChange(recorder record.EventRecorder, node *api.Node, new_s
 	// the time before we should try again
 	nextAttempt := time.Duration(0)
-	// have we deleted all pods
-	complete := true
-
-	selector := fields.OneTermEqualSelector(api.PodHostField, nodeName)
-	options := api.ListOptions{FieldSelector: selector}
-	pods, err := kubeClient.Core().Pods(api.NamespaceAll).List(options)
-	if err != nil {
-		return false, nextAttempt, err
-	}
-
+	// if the pod terminated
+	terminated := true
 	now := time.Now()
 	elapsed := now.Sub(since)
-	for _, pod := range pods.Items {
-		// Defensive check, also needed for tests.
-		if pod.Spec.NodeName != nodeName {
-			continue
-		}
-		// only clean terminated pods
-		if pod.DeletionGracePeriodSeconds == nil {
-			continue
-		}
 
-		// the user's requested grace period
-		grace := time.Duration(*pod.DeletionGracePeriodSeconds) * time.Second
-		if grace > maxGracePeriod {
-			grace = maxGracePeriod
-		}
+	// only clean terminated pod
+	if pod.DeletionGracePeriodSeconds == nil {
+		return terminated, nextAttempt, nil
+	}
 
-		// the time remaining before the pod should have been deleted
-		remaining := grace - elapsed
-		if remaining < 0 {
-			remaining = 0
-			glog.V(2).Infof("Removing pod %v after %s grace period", pod.Name, grace)
-			recordNodeEvent(recorder, nodeName, nodeUID, api.EventTypeNormal, "TerminatingEvictedPod", fmt.Sprintf("Pod %s has exceeded the grace period for deletion after being evicted from Node %q and is being force killed", pod.Name, nodeName))
-			if err := kubeClient.Core().Pods(pod.Namespace).Delete(pod.Name, api.NewDeleteOptions(0)); err != nil {
-				glog.Errorf("Error completing deletion of pod %s: %v", pod.Name, err)
-				complete = false
-			}
-		} else {
-			glog.V(2).Infof("Pod %v still terminating, requested grace period %s, %s remaining", pod.Name, grace, remaining)
-			complete = false
-		}
+	// the user's requested grace period
+	grace := time.Duration(*pod.DeletionGracePeriodSeconds) * time.Second
+	if grace > maxGracePeriod {
+		grace = maxGracePeriod
+	}
 
-		if nextAttempt < remaining {
-			nextAttempt = remaining
+	// the time remaining before the pod should have been deleted
+	remaining := grace - elapsed
+	if remaining < 0 {
+		remaining = 0
+		glog.V(2).Infof("Removing pod %v after %s grace period", pod.Name, grace)
+		recordNodeEvent(recorder, pod.Spec.NodeName, nodeUID, api.EventTypeNormal, "TerminatingEvictedPod", fmt.Sprintf("Pod %s has exceeded the grace period for deletion after being evicted from Node %q and is being force killed", pod.Name, pod.Spec.NodeName))
+		if err := kubeClient.Core().Pods(pod.Namespace).Delete(pod.Name, api.NewDeleteOptions(0)); err != nil {
+			glog.Errorf("Error completing deletion of pod %s: %v", pod.Name, err)
+			terminated = false
 		}
+	} else {
+		glog.V(2).Infof("Pod %v still terminating, requested grace period %s, %s remaining", pod.Name, grace, remaining)
+		terminated = false
 	}
-	return complete, nextAttempt, nil
+
+	if nextAttempt < remaining {
```

Comment:
Since you got rid of the loop, you don't really need this anymore, since nextAttempt is guaranteed to be 0 before this line. So I think you can get rid of this "if" clause, and the nextAttempt variable, and then change the return at the end of the function to 
```
return terminated, remaining, nil
```